Extension/src/LanguageServer/settings.ts

```diff
@@ -311,90 +312,205 @@ export class CppSettings extends Settings {
+    public get autoAddFileAssociations(): boolean { return this.getAsBoolean("autoAddFileAssociations"); }
+    public get workspaceParsingPriority(): string { return this.getAsString("workspaceParsingPriority"); }
+    public get workspaceSymbols(): string { return this.getAsString("workspaceSymbols"); }
+    public get exclusionPolicy(): string { return this.getAsString("exclusionPolicy"); }
+    public get refactoringIncludeHeader(): string { return this.getAsString("refactoring.includeHeader"); }
+    public get simplifyStructuredComments(): boolean { return this.getAsBoolean("simplifyStructuredComments"); }
+    public get doxygenGeneratedCommentStyle(): string { return this.getAsString("doxygen.generatedStyle"); }
+    public get doxygenGenerateOnType(): boolean { return this.getAsBoolean("doxygen.generateOnType"); }
+    public get commentContinuationPatterns(): (string | CommentPattern)[] {
+        // eslint-disable-next-line no-extra-parens
+        const value = super.Section.get<(string | CommentPattern)[]>("commentContinuationPatterns");
+        if (this.isArrayOfCommentContinuationPatterns(value)) {
+            return value;
+        }
+        const setting = getRawSetting("C_Cpp." + "commentContinuationPatterns");
+        return setting.default;
+    }
+    public get isConfigurationWarningsEnabled(): boolean { return this.getAsString("configurationWarnings").toLowerCase() === "enabled"; }
+    public get preferredPathSeparator(): string { return this.getAsString("preferredPathSeparator"); }
+    public get updateChannel(): string { return this.getAsString("updateChannel"); }
+    public get isVcpkgEnabled(): boolean { return this.getAsString("vcpkg").toLowerCase() === "enabled"; }
+    public get addNodeAddonIncludePaths(): boolean { return this.getAsBoolean("addNodeAddonIncludePaths"); }
+    public get renameRequiresIdentifier(): boolean { return this.getAsBoolean("renameRequiresIdentifier"); }
+    public get filesExclude(): Excludes { return this.getAsExcludes("files.exclude"); }
+    public get defaultIncludePath(): string[] { return this.getAsArrayOfStrings("default.includePath"); }
+    public get defaultDefines(): string[] { return this.getAsArrayOfStrings("default.defines"); }
+    public get defaultDotconfig(): string { return this.getAsString("default.dotConfig"); }
+    public get defaultMacFrameworkPath(): string[] { return this.getAsArrayOfStrings("default.macFrameworkPath"); }
+    public get defaultWindowsSdkVersion(): string { return this.getAsString("default.windowsSdkVersion"); }
+    public get defaultCompileCommands(): string { return this.getAsString("default.compileCommands"); }
+    public get defaultForcedInclude(): string[] { return this.getAsArrayOfStrings("default.forcedInclude"); }
+    public get defaultIntelliSenseMode(): string { return this.getAsString("default.intelliSenseMode"); }
+    public get defaultCompilerPath(): string { return this.getAsString("default.compilerPath") ?? undefined; }
```

Comment:
Should the return type of defaultCompilerPath be `string | undefined` The definition appears to `return ... ?? undefined`.

Reply:
The return type should be string | null as we don't want to return undefined here.

Reply:
I discussed this with @browntarik. This line was not supposed to change. It should be `?? undefined` because the purpose of the null is to delete a setting from a higher scope.